src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1069,7 +1069,7 @@ public DType ToRecord(ref bool fError)
                 case DKind.Record:
                     return this;
                 case DKind.LazyTable:
-                    return new DType(LazyTypeProvider, isTable: false, DisplayNameProvider);
+                    return new DType(LazyTypeProvider, isTable: false, DisplayNameProvider, AssociatedDataSources);
```

Comment:
> AssociatedDataSources) [](http://example.com/codeflow?start=91&length=23)

Are these gaps/bugs in current implementation?

Reply:
gaps in current implementation
discussed w/ Jas an Carlos who confirmed this is missing
have only able to repro an issue when using ServiceNow in NL repo, which uses a LazyRecord/Table